### PR TITLE
plugins/neogen: init + test

### PR DESCRIPTION
**Pull request by anriha**
Add a support for [neogen](https://github.com/danymat/neogen) plugin.

### CODE REVIEW

File: plugins/utils/neogen.nix [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        {
+            csharp = {
+                template = {
+                    annotation_convention = "...";
+                };
+            };
+        }
+        ```
+      '';
+      snippetEngine = helpers.defaultNullOpts.mkStr null ''
```
Comment (GaetanLepage) [inline plugins/utils/neogen.nix:90]:
```suggestion
      snippetEngine = helpers.mkNullOrOption types.str ''
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Very good !

Could you simply add a line break between each option declaration (to keep formatting coherent in the project) ?
Other than that, it looks good to me :)

File: plugins/utils/neogen.nix (hunk 1 above)
Comment (anriha) [reply plugins/utils/neogen.nix:90]: Missed that function, thanks. Btw, is there some documentation of these helper functions, or is the only option just going through the code?

Comment (GaetanLepage) [reply plugins/utils/neogen.nix]: Unfortunately, as the number of contributors has been fairly limited, we have not an amazing documentation for internal things.
However, you can have a look at the [CONTRIBUTING.md](https://github.com/pta2002/nixvim/blob/main/CONTRIBUTING.md) file. It gives a few information about the tools and the rules we use to package plugins.

**Comment by anriha** [comment]
Added the line breaks, hopefully it should be ok now.

**Review by GaetanLepage — APPROVED** [review]
I changed a few minor details.
Good to go !

**Pull request merged**